services/src/main/java/org/keycloak/utils/ReservedCharValidator.java

```diff
@@ -19,17 +19,20 @@
 import javax.ws.rs.BadRequestException;
 import org.jboss.logging.Logger;
 
+import java.util.Set;
+
 /**
  *
  * @author Stan Silvert
+ * @author Lukas Hanusovsky lhanusov@redhat.com
  */
 public class ReservedCharValidator {
     protected static final Logger logger = Logger.getLogger(ReservedCharValidator.class);
     
     // https://tools.ietf.org/html/rfc3986#section-2.2
-    private static final String[] RESERVED_CHARS = { ":", "/", "?", "#", "[", "@", "!", "$", 
+    private static final String[] RESERVED_CHARS = { ":", "/", "?", "#", "@", "!", "$",
                                                    "&", "(", ")", "*", "+", ",", ";", "=", 
-                                                   "]", "[", "\\" };
+                                                   "]", "[", "\\", "{", "}", "%" };
```

Comment:
This is not in line with https://tools.ietf.org/html/rfc3986#section-2.2 anymore. Is this intended? If so, needs to be justified in the comment above

Reply:
Thank you for your comments, I added there the comment with justification and also changed the Set into Iterable.

Reply:
I wonder if adding those three characters could break things for existing users?